// @wordpress/theme/build-module/theme-provider.mjs
// packages/theme/src/theme-provider.tsx
import { useMemo, useId } from "@wordpress/element";
import { ThemeContext } from "./context.mjs";
import { useThemeProviderStyles } from "./use-theme-provider-styles.mjs";

// packages/theme/src/style.module.css
var css = `.style-module__root__26kw6 {
	display: contents;
}
`;
document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(css));
var style_default = {
  "root": "style-module__root__26kw6"
};

// packages/theme/src/theme-provider.tsx
import { Fragment, jsx, jsxs } from "react/jsx-runtime";
function cssObjectToText(values) {
  return Object.entries(values).map(([key, value]) => `${key}: ${value};`).join("");
}
function generateCSSSelector({
  instanceId,
  isRoot
}) {
  const rootSel = `[data-wpds-root-provider="true"]`;
  const instanceIdSel = `[data-wpds-theme-provider-id="${instanceId}"]`;
  const selectors = [];
  if (isRoot) {
    selectors.push(
      `:root:has(.${style_default.root}${rootSel}${instanceIdSel})`
    );
  }
  selectors.push(`.${style_default.root}.${style_default.root}${instanceIdSel}`);
  return selectors.join(",");
}
var ThemeProvider = ({
  children,
  color = {},
  isRoot = false,
  density
}) => {
  const instanceId = useId();
  const { themeProviderStyles, resolvedSettings } = useThemeProviderStyles({
    color
  });
  const contextValue = useMemo(
    () => ({
      resolvedSettings
    }),
    [resolvedSettings]
  );
  return /* @__PURE__ */ jsxs(Fragment, { children: [
    themeProviderStyles ? /* @__PURE__ */ jsx("style", { children: `${generateCSSSelector({
      instanceId,
      isRoot
    })} {${cssObjectToText(themeProviderStyles)}}` }) : null,
    /* @__PURE__ */ jsx(
      "div",
      {
        "data-wpds-theme-provider-id": instanceId,
        "data-wpds-root-provider": isRoot,
        "data-wpds-density": density,
        className: style_default.root,
        children: /* @__PURE__ */ jsx(ThemeContext.Provider, { value: contextValue, children })
      }
    )
  ] });
};
export {
  ThemeProvider
};
